### PR TITLE
meta: remove code/test of unsupported versions

### PR DESCRIPTION
**Pull request by WikiRik**
## Pull Request Checklist

- [X] Have you added new tests to prevent regressions?
- [X] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [X] Did you update the typescript typings accordingly (if applicable)?
- [X] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [X] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

We had some additional code and tests for Node/database versions that we do not support in v7. This PR removes those. It does not update `defaultVersion` since those were higher than the removed code/tests.

### CODE REVIEW

File: test/integration/data-types/geometry.test.ts [hunk 1]
```diff
@@ -86,25 +85,6 @@ describe(getTestDialectTeaser('DataTypes'), () => {
     });
 
     it('correctly parses an empty GEOMETRY field', async () => {
```
Comment (WikiRik) [inline test/integration/data-types/geometry.test.ts:87]: Okay, this test seems to be broken.

---

For `mariadb` it indeed returns `null`.

For `mysql` it returns `Invalid GIS data provided to function st_geometryfromtext.`

For `postgres oldest` it returns `Too few ordinates in GeoJSON`

For `postgres latest` it returns the following;
```
{
  "coordinates": [
    NaN
    NaN
  ]
  "crs": {
    "properties": {
      "name": "EPSG:4326"
    }
    "type": "name"
  }
  "type": "Point"
}
```

**Comment by WikiRik** [comment]
Since the broken test is not in the scope of this PR I've skipped it and added a TODO

**Comment by WikiRik** [comment]
This PR should make #15346 a bit easier as well

**Comment by frostzt** [comment]
Ok this did make something come to my mind @ephys so in #15346 I actually changed initial version to `null`. The issue with this approach rises with tests especially unit tests that's why I did [this](https://github.com/sequelize/sequelize/pull/15346/files#diff-c791944fac0aba50f97b21713ea33d2f71679c834c9e87ffad79db3885874369). I don't know if that is a good approach!

**Review by ephys — APPROVED** [review]

**Review by ephys — APPROVED** [review]

**Review by frostzt — APPROVED** [review]

**Review by ephys — APPROVED** [review]

**Review by frostzt — APPROVED** [review]

**Pull request merged**